src/main/webapp/WEB-INF/static/js/locations.js

```diff
@@ -36,9 +36,9 @@ if (typeof (LocationsFunctions) === 'undefined') {
 		// function that prepares and initializes both the Select2 dropdown containing the location list as well as the checkbox that toggles between displaying only favorite
 		// locations or no. locationConversionFunction is provided as a parameter in case developers wish to change the construction of each select2 item, tho built-in function
 		// will be used by default if this is not provided
-		processLocationDropdownAndFavoritesCheckbox: function(locationSelectID, favoritesCheckboxID, allRadioButtonId, breedingLocationOnlyRadio, locationType, locationConversionFunction) {
+		processLocationDropdownAndFavoritesCheckbox: function(locationSelectID, favoritesCheckboxID, allRadioButtonId, breedingLocationOnlyRadio, locationType, locationConversionFunction, locationIdValue) {
 			LocationsFunctions.retrieveLocations().done(function(data) {
```

Comment:
Mmm the whole callback idea from the locations modal seems quite a bit convoluted to me, but It always has been like that. These changes add more complexity on top of it.
There's nothing much we can do for now, but this section really deserves a rewrite. In the end, it shouldn't be more complex than this:
```javascript
modal.open(...locationModal...).then(() => reloadLocationDropdown());
```